### DISCUSSION TITLE
RtcGetCalendarValue() return length

### DISCUSSION THREAD

**Discussion by 13868132273** [Lora-net/LoRaMac-node#1162] [Q&A]
Hello, I found that the same function RtcGetCalendarValue() return uint64_t when called by systime.c, but uint32_t when called by timer.c. Which is only a little more than 48 days(0xFFFFFFFF / 1024 = 4194303). 
Is there no problem with this use?
Thanks!

**Comment by mluis1** [comment]
The `RtcGetCalendarValue` function is defined `static` in the `rtc-board.c` driver. This means it is not directly accessible by code outside `rtc-board.c`.

The `RtcGetCalendarValue` is used to get the internal time reference and this why we need to have 64 bits variable to hold it.

For the outside world the time is provided using a 32 bits variable.
This was a design choice in order to not exclude processors that are not able to handle 64 bits variables.

We don't see any issue with such usage and limits.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.
